app/src/main/java/com/arctouch/codechallenge/home/data/repository/GenresRepository.kt

```diff
@@ -0,0 +1,30 @@
+package com.arctouch.codechallenge.home.data.repository
+
+import com.arctouch.codechallenge.core.api.TmdbApi
+import com.arctouch.codechallenge.core.domain.model.Genre
+import com.arctouch.codechallenge.home.data.cache.GenresCache
+
+class GenresRepository(private val api: TmdbApi) {
+
+    suspend fun getGenres(): List<Genre> {
+        return GenresCache.genres ?: getGenresFromApi()
+    }
+
+    private suspend fun getGenresFromApi(): List<Genre> {
+        return try {
+            val result = api.genresAsync().await()
+            val resultBody = result.body()
+            if (result.isSuccessful && resultBody != null) {
+                GenresCache.cacheGenres(resultBody.genres)
+                resultBody.genres
```

Comment:
I´am not sure, i don't see any clear improvement, it will do the same but with a different syntax, maybe if the result was nullable it could provide more safety